### PR TITLE
added Dockerfile

### PR DESCRIPTION
**Pull request by hershman**
This PR is a continuation of PR #326 which address issue #305 by adding a Dockerfile and documentation.

--
ACK: Just seeing docs on https://github.com/ga4gh/server/pull/337 about committing - sorry for the poorly named branch.

### CODE REVIEW

File: docs/source/installation.rst [hunk 1]
```diff
@@ -158,4 +158,19 @@ error log (in Apache on Debian/Ubuntu, for example, this is ``/var/log/apache2/e
 Deployment on Docker
 --------------------
 
-**TODO**
+It is also possible to deploy the server using Docker.
+
+- First build the Docker Image.  From the server/scripts folder run::
```
Comment (jeromekelleher) [inline docs/source/installation.rst]: This should be 'From the ``deploy`` directory run:'

Also, can you change the `::` pattern to use the code block syntax like the rest of the examples above please?

**Comment by jeromekelleher** [comment]
Works for me on Linux Mint 17.1 (I hadn't used Docker before, very impressive!). I think we'll want to refine this build process a bit over time, but this is a great start.

+1 modulo tiny niggles.

File: docs/source/installation.rst [hunk 2]
```diff
@@ -158,4 +158,25 @@ error log (in Apache on Debian/Ubuntu, for example, this is ``/var/log/apache2/e
 Deployment on Docker
 --------------------
 
-**TODO**
+It is also possible to deploy the server using Docker.
+
+First build the Docker Image.  From the ``deploy/scripts`` folder run:
```
Comment (jeromekelleher) [inline docs/source/installation.rst]: Shouldn't this just be ``deploy``?

**Comment by jeromekelleher** [comment]
Looks good @hershman. Can you fix the small bug and squash the commits please?

I think we're good to merge then (unless @macieksmuga or @dcolligan have any changes they'd like to see?).

**Comment by dcolligan** [comment]
Yeah, after squashing the commits and rebasing on top of the latest develop (to get rid of the Travis failure) this can be merged.

**Comment by afirth** [comment]
@hershman Could you take a look at #350 if you get a chance?

**Comment by jeromekelleher** [comment]
Can you squash the commits and update this PR please @hershman? I think it's ready to merge...

**Comment by afirth** [comment]
I brought @hershman's Dockerfile (with the addition of author info) into #350 as a variant and set up autobuilds. Greatly expanded the docs. Waiting for the builds to complete before I squash it. The Dockerfile in #350 supports using a local dataset rather than being restricted to the demo and is based on the Dockerfile from the mod_wsgi author

**Comment by hershman** [comment]
I squashed down the commits - thanks for the help @jeromekelleher .

**Comment by jeromekelleher** [comment]
Merged - thanks @hershman!

**Pull request merged**